### ISSUE TITLE
Allow `Schema.definitions` to be a file

### ISSUE THREAD

**Issue by dblythy** [parse-community/parse-server#8395]
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Current Limitation

I like to store a JSON file with the SCHEMA definition. It could be nice to be able to set definitions via:

```
 schema: {
  definitions: 'schema.json'
},
```

### Feature / Enhancement Description

Allow setting a file to schema definitions

### Example Use Case

```
 schema: {
  definitions: 'schema.json'
},
```

### Alternatives / Workarounds

```js
const definitions = await fs
    .readFile('spec/support/schema.json', 'utf8')
    .then(file => JSON.parse(file));
schema: {
  definitions
},
```

**Comment by sadortun** [comment]
Using JSON would require additional validation to ensure the loaded file is valid.

It would also add potential issues at run-time instead of compile time.  if the JSON file is invalid or missing.

Considering the workaround is 3 lines, I'm not sure if it's a good thing to add complexity on the server for this.

**Comment by mtrezza** [comment]
> Using JSON would require additional validation to ensure the loaded file is valid.

`JSON.parse()` throws an error if the input is an invalid JSON, so there is no additional validation needed that is specific to reading from file.

> It would also add potential issues at run-time instead of compile time. if the JSON file is invalid or missing.

There is no compilation step in JavaScript, everything happens at run-time. The same argument could be made for HTML files that are loaded for password reset, or the localization feature of Pages.

> Considering the workaround is 3 lines, I'm not sure if it's a good thing to add complexity on the server for this.

Adding 3 lines doesn't seem to add much complexity?

**Comment by sadortun** [comment]
> JSON.parse() throws an error if the input is an invalid JSON

Yes, but it doesn't validate the content of the JSON itself ( compared to Typescript or Flow/JSDoc ), not sure loading potential invalid JSON file at runtime without explicit validation is safe.

> There is no compilation step in JavaScript, everything happens at run-time.

Sorry, I forgot old JavaScript still exists. 🤣 Typescript/Flow will throw errors at you if you try to include an invalid schema definition. 🤣 + static analysis tools

> Adding 3 lines doesn't seem to add much complexity?

Adding features to the `parse-server` that can't be done entirely on the user side with boilerplate code  is probably something that can remain on the user side. 

I'm just not sure what core issue adding this would solve. It's just my 2 cents anyway 🥲

Also, You will need to add validation for the JSON file itself. Plus write+maintain documentation that will explain both way of doing things.

**Comment by mtrezza** [comment]
> Yes, but it doesn't validate the content of the JSON itself

What kind of validation? I'd say validation should be source agnostic; whether the schema comes from a file or as a Parse Server option shouldn't matter.
